staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -204,6 +207,10 @@ func (r *proxyHandler) updateAPIService(apiService *apiregistrationapi.APIServic
 		serviceName:      apiService.Spec.Service.Name,
 		serviceNamespace: apiService.Spec.Service.Namespace,
 	}
-	newInfo.proxyRoundTripper, newInfo.transportBuildingError = restclient.TransportFor(newInfo.restConfig)
+	if r.proxyTransport == nil {
+		newInfo.proxyRoundTripper, newInfo.transportBuildingError = restclient.TransportFor(newInfo.restConfig)
+	} else {
+		newInfo.proxyRoundTripper, newInfo.transportBuildingError = r.proxyTransport, nil
```

Comment:
I am not convinced this will cause the use of the key/cert data, considering what restclient.TransportFor seems to do: https://github.com/kubernetes/kubernetes/blob/1a46a167f3c1baecb7e0e6a84137330982861ec2/staging/src/k8s.io/client-go/transport/transport.go#L79